### ISSUE TITLE
UI build fails with "The re-exported type org.eclipse.swt.ole.win32.OLE has been removed from org.eclipse.jface_3.32.0"

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-platform/eclipse.platform.ui#1654]
See https://github.com/eclipse-platform/eclipse.platform.ui/pull/1653 and https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-1653/1/console

```
10:25:14.707 [ERROR] Failed to execute goal org.eclipse.tycho:tycho-apitools-plugin:4.0.6-SNAPSHOT:verify (verify) on project org.eclipse.jface: There are API errors:
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OLE has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleAutomation has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleClientSite has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleControlSite has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleEvent has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleFrame has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleFunctionDescription has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleListener has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OleParameterDescription has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.OlePropertyDescription has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:0 The re-exported type org.eclipse.swt.ole.win32.Variant has been removed from org.eclipse.jface_3.32.0
10:25:14.707 [ERROR] META-INF/MANIFEST.MF:5 The major version should be incremented in version 3.33.0, since API breakage occurred since version 3.32.0
```

**Comment by iloveeclipse** [comment]
`The re-exported type org.eclipse.swt.ole.win32` - I don't see why PDE API complains about a package that was never exported by SWT main bundle? It was and is only part of native fragment on Windows.

**Comment by laeubi** [comment]
> `The re-exported type org.eclipse.swt.ole.win32` - I don't see why PDE API complains about a package that was never exported by SWT main bundle? It was and is only part of native fragment on Windows.

Seems like API tools does not consider Platfom filters correctly.

**Comment by iloveeclipse** [comment]
@HannesWell : I wonder if https://github.com/eclipse-pde/eclipse.pde/pull/1094 helps in any way here as it seem to remove SWT special handling from PDE API tooling? Or rather we should add something additional to avoid SWT API from different platform be checked?

**Comment by iloveeclipse** [comment]
> Seems like API tools does not consider Platfom filters correctly.

This must be recently introduced or caused by recent SWT project or build restructuring.

**Comment by laeubi** [comment]
I'll take a look if it could be fixed on Tycho side assuming we dont see this in the IDE?

**Comment by iloveeclipse** [comment]
Thanks. Is this Tycho change visible immediately, or do we need to wait some days to see the fixed Tycho build?

**Comment by HannesWell** [comment]
> Thanks. Is this Tycho change visible immediately, or do we need to wait some days to see the fixed Tycho build?

I just submitted the backport and its build should be completed in about an hour or two or less (don't know the exact runtimes).

Then it depends how fast the verification builds update the snapshot dependencies/plugins. The latest it should be every 24h, but if `-U` is passed it should be immediately.

Regarding the cause, Christoph can probably tell best, but to me it looks like the enhancement from https://github.com/eclipse-tycho/tycho/pull/3457 in combination with the re-export of SWT by jface let the APItools find more packages in the baseline than actually exist.

**Comment by laeubi** [comment]
> Thanks. Is this Tycho change visible immediately, or do we need to wait some days to see the fixed Tycho build?

Here is the build job of the tycho-4 line: https://ci.eclipse.org/tycho/job/tycho-github/job/tycho-4.0.x/

As you can see this specific fix is/was available from 07.02.2024, 19:02 CEST + roughly 2.5hrs
